datasketches-memory-java8/src/main/java/org/apache/datasketches/memory/internal/BaseStateImpl.java

```diff
@@ -446,4 +442,41 @@ static final String toHex(final BaseStateImpl state, final String preamble, fina
     return sb.toString();
   }
 
+  //MONITORING
+
+  /**
+   * Gets the current number of active direct memory allocations.
+   * @return the current number of active direct memory allocations.
+   */
+  public static final long getCurrentDirectMemoryAllocations() {
+    return BaseStateImpl.currentDirectMemoryAllocations_.get();
+  }
+
+  /**
+   * Gets the current size of active direct memory allocated.
+   * @return the current size of active direct memory allocated.
+   */
+  public static final long getCurrentDirectMemoryAllocated() {
+    return BaseStateImpl.currentDirectMemoryAllocated_.get();
+  }
+
+  /**
+   * Gets the current number of active direct memory map allocations.
+   * @return the current number of active direct memory map allocations.
+   */
+  public static final long getCurrentDirectMemoryMapAllocations() {
+    return BaseStateImpl.currentDirectMemoryMapAllocations_.get();
+  }
+
+  /**
+   * Gets the current size of active direct memory map allocated.
+   * @return the current size of active direct memory map allocated.
+   */
+  public static final long getCurrentDirectMemoryMapAllocated() {
+    return BaseStateImpl.currentDirectMemoryMapAllocated_.get();
+  }
```

Comment:
What is the difference between allocated vs allocations?  Is the latter some kind of rate measurement?

Reply:
Allocated: the amount of total off-heap memory allocated via the WritableMemory.allocateDirect(..) mechanism.
Allocations: The number of allocation requests.